### PR TITLE
Task J-75. Add current user id as trace id in logger

### PR DESCRIPTION
**Pull request by Denidzio**

### CODE REVIEW

File: backend/src/middlewares/set-trace-id/set-trace-id.middleware.ts [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+import { appAsyncStorage, token } from '~/services/services';
 import { getRandomId } from '~/helpers/helpers';
 import { AppAsyncStorageKey } from '~/common/enums/enums';
+import { TokenPayload } from '~/common/types/types';
 
-const setTraceId: RequestHandler = async (_req, _res, next) => {
-  const traceId = getRandomId();
+const setTraceId: RequestHandler = async (req, _res, next) => {
+  const authorizationHeader = req.get('Authorization');
+  const bearerToken = authorizationHeader?.split(' ')[1];
```
Comment (ghost) [inline backend/src/middlewares/set-trace-id/set-trace-id.middleware.ts]: what does 1 mean?

we already have this functionality in the current-user route

we need reuse it

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by Denidzio** [comment]
[Add current user to `trace-id` in the logger middleware](https://trello.com/c/SQHzaLW3/75-add-current-user-to-trace-id-in-the-logger-middleware)

File: backend/src/helpers/jwt/extract-jwt-token/extract-jwt-token.helper.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const extractJwtToken = (header: string): string => {
+  const [, token] = header.split(' ');
+  return token;
```
Comment (ghost) [inline backend/src/helpers/jwt/extract-jwt-token/extract-jwt-token.helper.ts]:
```suggestion
  const [, token] = header.split(' ');
  return token ?? null;
```

File: backend/src/helpers/jwt/extract-jwt-token/extract-jwt-token.helper.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const extractJwtToken = (header: string): string => {
```
Comment (ghost) [inline backend/src/helpers/jwt/extract-jwt-token/extract-jwt-token.helper.ts]:
```suggestion
const extractAuthToken = (header: string): string => {
```
don't forget to change the name of the file

**Review by ghost — CHANGES_REQUESTED** [review]

File: backend/src/helpers/jwt/extract-auth-token/extract-auth-token.helper.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const extractAuthToken = (header: string | undefined): string | null => {
```
Comment (ghost) [inline backend/src/helpers/jwt/extract-auth-token/extract-auth-token.helper.ts]:
```suggestion
const extractAuthToken = (header?: string): string | null => {
```

File: backend/src/helpers/jwt/jwt.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export { extractAuthToken } from './extract-auth-token/extract-auth-token.helper';
```
Comment (ghost) [inline backend/src/helpers/jwt/jwt.ts]: backend/src/helpers/token/token.ts

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by yevhenprybytok — APPROVED** [review]

**Pull request merged**